src/main/java/com/faforever/client/replay/ReplayService.java

```diff
@@ -201,8 +201,21 @@ private void onLocalReplaysWatchEvent(WatchKey key) {
       }
     }
 
-    localReplays.addAll(newReplays);
-    publisher.publishEvent(new LocalReplaysChangedEvent(this, newReplays, deletedReplays));
+    CompletableFuture[] replayFuturesArray = newReplaysFutures.toArray(new CompletableFuture[newReplaysFutures.size()]);
+    CompletableFuture<List<Replay>> newReplaysFuture = CompletableFuture.allOf(replayFuturesArray)
+        .thenApply(ignoredVoid ->
+            newReplaysFutures.stream()
+                .map(future -> future.join())
```

Comment:
use method reference
```suggestion
                CompletableFuture::join
```